src/minimetrics/core.py

```diff
@@ -187,64 +213,57 @@ def _hash(x: Any) -> int:
 
 class Aggregator:
     ROLLUP_IN_SECONDS = 10.0
-    MAX_COMPLEXITY = 100000
+    MAX_WEIGHT = 100000
 
     def __init__(self) -> None:
-        self.buckets: Dict[ComposedKey, Metric[Any]] = {}
-        self._bucket_complexity: int = 0
+        self.buckets: Dict[BucketKey, Metric[Any]] = {}
+        # Stores the total weight of the in-memory buckets. Weight is determined on a per metric type basis and
+        # represents how much weight is there to represent the metric (e.g., counter = 1, distribution = n).
```

Comment:
_nit_: This is a recursive definition. As an idea, what if you describe it as "how many data points this metric carries"?